core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -655,7 +656,16 @@ public void setDefaultTTL(int defaultTTL) {
                         .doOnNext(i -> insertedDataPointEvents.onNext(metric)))
                 .doOnNext(meter::mark);
 
-        Observable<Integer> indexUpdates = dataAccess.updateMetricsIndex(metrics)
+        Observable<Metric<T>> updateIdxMetrics = metrics.filter(metric -> {
```

Comment:
All of the index related work can/should be done out of band. We want to treat as an async, background task. This means a couple key changes. I would change the return statement from 

```return updates.mergeWith(indexUpdates)```

to just

```return updates```

And then we will need to subscribe to ```indexUpdates`` (or whatever index related observables we're dealing with) separately so that the work does get performed.